generator/src/routes/+page.svelte

```diff
@@ -98,12 +103,29 @@
     document.body.removeChild(link)
   }
 
-  function onChangeValidSignature(event: Event) {
-    const target = event.target as HTMLInputElement
-    if (target.checked) {
+  function clearError() {
+    error = null
+  }
+  function onChangeValidSignature() {
+    if (isValid) {
       issValue = settings.ISS_DOMAIN
     }
   }
+  function onChangeProductType() {
+    productId = ""
+    product = ""
+  }
+  function onChangeProductId() {
+    if (productType === ProductType.PREMADE) {
+      product = premadeProducts.find((product) => product.id === productId)?.product!
+    }
```

Comment:
This logic seems unclear and the name for this function is not great - the dropdown is now "product id", but there is a separate "product id" field. Just seeing "onChangeProductId()" does not in any way to me hint that this has anything to do with selecting a premade product from the dropdown

You should probably call the dropdown "premade product" or something and when it is changed pick the values from it. Also I would rather use "name" as the unique identifier for those entries in this case.